spring-boot-autoconfigure/src/test/java/org/springframework/boot/autoconfigure/orm/jpa/AbstractJpaAutoConfigurationTests.java

```diff
@@ -77,10 +80,40 @@ public void close() {
 	public void testNoDataSource() throws Exception {
 		this.context.register(PropertyPlaceholderAutoConfiguration.class,
 				getAutoConfigureClass());
-		this.expected.expect(BeanCreationException.class);
-		this.expected.expectMessage("No qualifying bean");
-		this.expected.expectMessage("DataSource");
 		this.context.refresh();
+		assertThat(this.context.getBeansOfType(JpaProperties.class).size()).isEqualTo(1);
```

Comment:
Not sure I like that, you're changing the semantic a lot (and it's wider than the intent of this PR IMO).